### PR TITLE
for py3 tests

### PR DESCRIPTION
**Pull request by nickpell**
From https://github.com/dimagi/commcare-hq/pull/19116

### CODE REVIEW

**Comment by nickpell** [comment]
@dimagi/py3

File: corehq/apps/app_manager/models.py [hunk 1]
```diff
@@ -2218,8 +2218,10 @@ def wrap(cls, data):
 
         # Lazy migration: enum used to be a dict, now is a list
         if isinstance(data.get('enum'), dict):
-            data['enum'] = sorted({'key': key, 'value': value}
-                                  for key, value in data['enum'].items())
+            data['enum'] = sorted(
+                [{'key': key, 'value': value} for key, value in data['enum'].items()],
```
Comment (emord) [inline corehq/apps/app_manager/models.py]: I think you need `six.items` to not get yelled at by modernize next week

File: corehq/apps/es/aggregations.py [hunk 2]
```diff
@@ -193,7 +193,7 @@ def doc_count(self):
         return self.result['doc_count']
 
     def __getattr__(self, attr):
-        sub_aggregation = filter(lambda a: a.name == attr, self.aggregations)[0]
+        sub_aggregation = list(filter(lambda a: a.name == attr, self.aggregations))[0]
```
Comment (emord) [inline corehq/apps/es/aggregations.py]: should probably be rewritten as:

`next(agg for agg in self.aggregations if agg.name == attr)`

https://stackoverflow.com/a/2364277

File: corehq/apps/export/models/new.py [hunk 3]
```diff
@@ -2490,10 +2490,10 @@ def get_value(self, domain, doc_id, doc, base_path, transform_dates=False, **kwa
         new_doc = None
         if isinstance(value, list):
             try:
-                new_doc = filter(
+                new_doc = list(filter(
```
Comment (emord) [inline corehq/apps/export/models/new.py]: Same `next` rewrite possible here

**Review by emord — COMMENTED** [review]
I don't think I remember enough about the encode/decode semantics to sign off on those. looks like there are quite a few of those in real code in this pr. will let someone else approve those (or can re-aquatint myself with it if no one else is available)

**Comment by nickpell** [comment]
Encode/decode works as follows:

```
>>> b'text'.decode('utf-8')
u'text'
>>> u'text'.encode('utf-8')
b'text'
```

I realize encode/decode changes are particularly hard to tell if they're correct, but all of these changes should have test coverage, so I'm pretty confident in them.

File: corehq/apps/app_manager/tests/util.py [hunk 4]
```diff
@@ -113,6 +113,10 @@ def parse_normalize(xml, to_string=True, is_html=False):
 
 def _check_shared(expected, actual, checker, extension):
     # snippet from http://stackoverflow.com/questions/321795/comparing-xml-in-a-unit-test-in-python/7060342#7060342
+    if isinstance(expected, six.binary_type):
```
Comment (millerdev) [inline corehq/apps/app_manager/tests/util.py]: nit: was this an automatic change by a modernizer script? If not `bytes` means the same thing on both Python 2 and 3, so is preferable to `six.binary_type`.

**Review by millerdev — APPROVED** [review]

File: corehq/apps/app_manager/tests/util.py (hunk 4 above)
Comment (nickpell) [reply corehq/apps/app_manager/tests/util.py]: This was a manual change.  Good point about ```six.binary_type``` - I'll be conscious of that going forward, and do a cleanup at some point.

Comment (nickpell) [reply corehq/apps/app_manager/tests/util.py]: https://trello.com/c/RUI4xVF3/68-sixbinarytype-bytes

**Pull request merged**